frontend/screens/chat/ChatInterface.js

```diff
@@ -7,79 +7,93 @@ import {
+    await getMessages(conversationId, userId, userType, setMessages);
+    }
+    else
+      showAlert("Error","Please provide proper message")
   }
   return (
     <View style={styles.container}>
-      <ScrollView contentContainerStyle={styles.scrollContentContainer}>
+      <TouchableOpacity
+        style={styles.topbar}
+        onPress={() => navigation.navigate("InboxInterface")}
+      >
+        <Depth1Frame7
+          depth4Frame0={require("../../assets/depth-4-frame-017.png")}
+          requestDetails={`Chat with ${name}`}
+          depth3Frame0BackgroundColor="#fff"
+          requestDetailsWidth={"auto"}
+          depth4Frame0FontFamily="BeVietnamPro-Bold"
+          depth4Frame0Color="#000"
+        />
+      </TouchableOpacity>
+      <ScrollView ref={scrollView} contentContainerStyle={styles.scrollContentContainer} onContentSizeChange={() => scrollView.current?.scrollToEnd()}>
         <View style={styles.chatContent}>
-          <TouchableOpacity
-            onPress={() => navigation.navigate("InboxInterface")}
-          >
-            <Depth1Frame7
-              depth4Frame0={require("../../assets/depth-4-frame-017.png")}
-              requestDetails={`Chat with ${name}`}
-              depth3Frame0BackgroundColor="#fff"
-              requestDetailsWidth={"auto"}
-              depth4Frame0FontFamily="BeVietnamPro-Bold"
-              depth4Frame0Color="#000"
-            />
-          </TouchableOpacity>
           <View style={styles.messagesContainer}>
             {messages.map((message, index) => (
-              <View key={index} style={styles.messageRow}>
-                <View style={styles.messageLine}>
-                  <View style={[styles.messageDot, styles.lineHeight]} />
-                  <View style={styles.messageSmallDot} />
-                  <View style={[styles.messageDot, styles.lineHeight]} />
-                </View>
-                <View style={styles.messageContent}>
-                  <View style={styles.messageTextContainer}>
-                  {message.sender.name}
-                    <View style={styles.messageTextWrapper}>
-                      <Text style={[styles.messageText, styles.timeText]}>
-                        {message.content}
-                      </Text>
-                    </View>
-                  </View>
-                  <View style={styles.messageTimeContainer}>
-                    <View style={styles.messageTextContainer}>
-                      <Text style={[styles.timeAgo, styles.timeText]}>
-                        {message.timeAgo}
-                      </Text>
-                    </View>
-                  </View>
-                </View>
-              </View>
+              message.sender.name=="You"?
+              <SenderMessage key={index} name={message.sender.name} content={message.content} timeAgo={message.timeAgo}/>:
+              <ReceiverMessage key={index} name={message.sender.name} content={message.content} timeAgo={message.timeAgo}/>
```

Comment:
try to use optional check here so that it doesn't break the feature like `message?.sender?.name`